src/text-prompt.js

```diff
@@ -3,41 +3,64 @@ import { EOL } from "node:os";
 
 // Import Third-party Dependencies
 import ansi from "ansi-styles";
+import stripAnsi from "strip-ansi";
 
 // Import Internal Dependencies
 import { AbstractPrompt } from "./abstract-prompt.js";
 import { SYMBOLS } from "./constants.js";
 
 export class TextPrompt extends AbstractPrompt {
-  #question;
+  #validators;
 
-  constructor(message, stdin = process.stdin, stdout = process.stdout) {
+  constructor(message, options = {}) {
+    const { stdin = process.stdin, stdout = process.stdout, validators = [] } = options;
     super(message, stdin, stdout);
 
-    this.#question = new Promise((resolve) => {
-      this.rl.question(`${ansi.bold.open}${SYMBOLS.QuestionMark} ${message}${ansi.bold.close} `, (answer) => {
+    this.#validators = validators;
+    this.questionPrefix = `${ansi.bold.open}${SYMBOLS.QuestionMark} `;
+    this.questionSuffix = `${ansi.bold.close} `;
+    this.questionSuffixError = "";
+  }
+
+  #question() {
+    return new Promise((resolve) => {
+      const question = `${this.questionPrefix}${this.message}${this.questionSuffix}${this.questionSuffixError}`;
```

Comment:
I saw that we have a few complicated template strings. WDYT about creating methods for these?

Reply:
Sure, i've simplified theses strings and some little things with more methods